internal/internal_worker.go

```diff
@@ -923,14 +918,16 @@ func initBinaryChecksumLocked() error {
 }
 
 func getBinaryChecksum() string {
-	binaryChecksumLock.Lock()
-	defer binaryChecksumLock.Unlock()
+	binaryChecksumLock.RLock()
+	val := binaryChecksum
+	binaryChecksumLock.RUnlock()
 
-	if len(binaryChecksum) == 0 {
-		err := initBinaryChecksumLocked()
-		if err != nil {
-			panic(err)
-		}
+	if len(val) != 0 { // already initialized.
+		return val
+	}
+
+	if err := initBinaryChecksum(); err != nil {
+		panic(err)
 	}
 
 	return binaryChecksum
```

Comment:
this is no longer protected, and is a racy read.
a simple/safe fix may be to just return the calculated value from init.

or switch to atomics, and don't care about races between concurrent inits - any value is equally valid (and they are likely identical).

Reply:
Good catch. Changing type to `binaryChecksum atomic.Value` and getting rid of mutexes for simplicity. Given the usage pattern it seems fine to init multiple times if it happens to be called concurrently during startup